src/Orleans/Runtime/AsynchQueueAgent.cs

```diff
@@ -30,7 +30,10 @@ public void QueueRequest(T request)
                 queueTracking.OnEnQueueRequest(1, requestQueue.Count, request);
             }
 #endif
-            requestQueue.Add(request);
+            if (requestQueue != null)
+            {
+                requestQueue.Add(request);
```

Comment:
I invert it, but how is it better? for readability?

The reason from my side was that since this is the last statement in that method an early return doesn't benefit anything.

I'm moving the null check to the beginning to not to execute the stats code in case the ```requestQueue``` is null, because that can falsify the statistics ;-)